common/util/autoware_utils/include/autoware_utils/trajectory/convert.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE_UTILS__TRAJECTORY__CONVERT_HPP_
+#define AUTOWARE_UTILS__TRAJECTORY__CONVERT_HPP_
+
+#include "autoware_utils/geometry/geometry.hpp"
+#include "autoware_utils/geometry/pose_deviation.hpp"
+
+#include "autoware_auto_planning_msgs/msg/trajectory.hpp"
+#include "autoware_auto_planning_msgs/msg/trajectory_point.hpp"
+
+#include <boost/optional.hpp>
+
+#include <algorithm>
+#include <vector>
+
+namespace autoware_utils
+{
+using autoware_auto_planning_msgs::msg::Trajectory;
+using TrajectoryPointArray = std::vector<autoware_auto_planning_msgs::msg::TrajectoryPoint>;
+
+// convertToTrajectoryByClipping() just clips TrajectoryPointArray up to the capacity of Trajectory.
+// Therefore, the error handling out of this function is necessary if the size of
+// TrajectoryPointArray greater than the capacity.
+// (TODO) Decide how handle the situation that we need to use the trajectory with size of points
+// larger than the capacity.
+Trajectory convertToTrajectoryByClipping(const TrajectoryPointArray & trajectory)
```

Comment:
Please add the comments like the following.
- This function is temporarily added for porting to autoware_auto_msgs.
- We should consider whether to remove this function after the porting is done.

Reply:
Also, I think it's good to rename the file name to `tmp_trajectory_conversion.hpp` or something like that.